pkg/objectupdate/rte/rte.go

```diff
@@ -136,6 +136,14 @@ func DaemonSetArgs(ds *appsv1.DaemonSet, conf nropv1.NodeGroupConfig) error {
 		return fmt.Errorf("cannot modify the arguments for container %s", cnt.Name)
 	}
 
+	rtePauseEnabled := isRTEPauseEnabled(&conf)
+	klog.V(2).InfoS("DaemonSet update: RTEPause status", "daemonset", ds.Name, "enabled", rtePauseEnabled)
+	if rtePauseEnabled {
+		flags.SetToggle("--no-publish")
+	} else if _, ok := flags.GetFlag("--no-publish"); ok {
```

Comment:
`flagcodec` should allow you to remove the flag directly, without check+delete. Please check

Reply:
indeed, it already checks that internally. 
https://github.com/k8stopologyawareschedwg/deployer/blob/main/pkg/flagcodec/flagcodec.go#L94C3-L94C3
thanks for pointing this out.